smarts/core/waypoints.py

```diff
@@ -111,16 +112,28 @@ def __eq__(self, other):
         )
 
 
-LinkedWaypoint = namedtuple(
-    "LinkedWaypoint",
-    [
-        "wp",  # Waypoint: current waypoint
-        "nexts",  # list of LinkedWaypoint: list of next immediate waypoints
+class LinkedWaypoint:
+    def __init__(self, wp=None, nexts=[], is_shape_wp=False):
+        self.wp = wp  # Waypoint: current waypoint
+        self.is_shape_wp = is_shape_wp
+        self.nexts = nexts  # list of LinkedWaypoint: list of next immediate waypoints
```

Comment:
The following is a suggestion for the entire `LinkedWaypoint` class. Unfortunately, I could not highlight the entire class while commenting. 

I believe it is better to maintain the immutable data type, namely `NamedTuple`, with type hints. It can be done as follows:

```suggestion
class LinkedWaypoint(NamedTuple):
    wp: Waypoint = None  # Waypoint: current waypoint
    nexts: List[Waypoint] = [] # list of LinkedWaypoint: list of next immediate waypoints
    is_shape_wp: bool = False

    def __hash__(self):
        ## distinguish between different continuations here too
        ## so the lru_cache on _waypoints_starting_at_waypoint() below
        ## doesn't return the wrong set of waypoints.
        return hash(self.wp) + sum(hash(nwp.wp) for nwp in self.nexts)
```

Reply:
done.

Reply:
done.